src/NuGetPackageVerifier/build/NuGetPackageVerifier.targets

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Target Name="VerifyPackages"
+          AfterTargets="Pack"
```

Comment:
> instead of having executables that get run, would it make sense to just these be MSBuild tasks?

Yes,I was thinking the same thing too. It will take more effort to port this to a task, and one we'll have to do carefully. I expect we will run into assembly loading issues with MSBuild for .NET Core. It's default load context may already have NuGet assemblies.

> Having the lifecycle spread out between the build project + individual tools might be harder to manage and tedious to debug if they never get invoked.

A reasonable concern, but one I'm sure we have the brainpower to tackle. There are already some good tools for debugging MSBuild. https://github.com/KirillOsenkov/MSBuildStructuredLog is a good example.